### PR TITLE
[TM] Chp3-1 More inputs to the vector attribute data section

### PR DESCRIPTION
**Pull request by DelazJ**
In [chapter 4](https://docs.qgis.org/testing/en/docs/training_manual/vector_classification/attribute_data.html#basic-fa-attribute-data) we refer to this section for some background knowledge. However, there's not so much food in it. This PR tries to teach some more basic concepts and adds an exercise.

### CODE REVIEW

File: source/docs/training_manual/answers/answers.rst [hunk 1]
```diff
@@ -58,6 +58,29 @@ Refer back to the :ref:`browser_panel_tm` and to :ref:`load_geopackage` sections
...
 
+.. _vector-explore-attribute-data:
+
+|basic| *Attribute data*
+...............................................................................
+
+* There should be 9 fields in the :guilabel:`rivers` layer:
+
+  #. Select the layer in the :guilabel:`Layers` panel.
+  #. Right-click and choose :guilabel:`Attribute table`, or press the |openTable|
```
Comment (havatv) [inline source/docs/training_manual/answers/answers.rst]:
```suggestion
  #. Right-click and choose :guilabel:`Open Attribute Table`, or press the |openTable|
```

File: source/docs/training_manual/answers/answers.rst [hunk 2]
```diff
@@ -58,6 +58,29 @@ Refer back to the :ref:`browser_panel_tm` and to :ref:`load_geopackage` sections
...
+     button on the :guilabel:`Attributes Toolbar`.
+  #. Count the number of columns.
+
+  .. tip:: A quicker approach could also consist in double-clicking the
```
Comment (havatv) [inline source/docs/training_manual/answers/answers.rst]:
```suggestion
  .. tip:: A quicker approach could be to double-click the
```

File: source/docs/training_manual/answers/answers.rst [hunk 3]
```diff
@@ -58,6 +58,29 @@ Refer back to the :ref:`browser_panel_tm` and to :ref:`load_geopackage` sections
...
+   :guilabel:`rivers` layer, open :menuselection:`Layer properties --> Source
```
Comment (havatv) [inline source/docs/training_manual/answers/answers.rst]:
```suggestion
   :guilabel:`rivers` layer, open the :menuselection:`Layer properties --> Source
```

File: source/docs/training_manual/answers/answers.rst [hunk 4]
```diff
@@ -58,6 +58,29 @@ Refer back to the :ref:`browser_panel_tm` and to :ref:`load_geopackage` sections
...
+   Fields` tab and pick there the count of fields.
```
Comment (havatv) [inline source/docs/training_manual/answers/answers.rst]:
```suggestion
   Fields` tab and count of fields.
```

File: source/docs/training_manual/answers/answers.rst [hunk 5]
```diff
@@ -58,6 +58,29 @@ Refer back to the :ref:`browser_panel_tm` and to :ref:`load_geopackage` sections
...
+
+* Information about towns are available in the :guilabel:`places` layer. Open its
+  attribute table as previously seen: there are two features whose :guilabel:`place`
```
Comment (havatv) [inline source/docs/training_manual/answers/answers.rst]:
```suggestion
  attribute table (in the same way as for the:guilabel:`rivers` layer): there are two features whose :guilabel:`place`
```

File: source/docs/training_manual/answers/answers.rst [hunk 6]
```diff
@@ -58,6 +58,29 @@ Refer back to the :ref:`browser_panel_tm` and to :ref:`load_geopackage` sections
...
+  attribute is set to ``town``: *Swellendam* and *Buffeljagsrivier*.
+  You can add any more information picked from these two records.
```
Comment (havatv) [inline source/docs/training_manual/answers/answers.rst]: I don't understand this sentence.  Should be removed or reformulated.

File: source/docs/training_manual/basic_map/vector_data.rst [hunk 7]
```diff
@@ -8,9 +8,10 @@
 ===============================================================================
 
 Vector data is arguably the most common kind of data you will find in the daily
-use of GIS. It describes geographic data in terms of points, that may be
-connected into lines and polygons. Every object in a vector dataset is called a
-**feature**, and is associated with data that describes that feature.
+use of GIS. It describes geographic data in terms of points, lines and polygons on a
```
Comment (havatv) [inline source/docs/training_manual/basic_map/vector_data.rst]:
```suggestion
use of GIS. The 2D vector model describes the location and shape of geographic features using points, lines and polygons.
```

File: source/docs/training_manual/basic_map/vector_data.rst [hunk 8]
```diff
@@ -8,9 +8,10 @@
...
+coordinate plane. It is usually used to store discrete features, like roads and
```
Comment (havatv) [inline source/docs/training_manual/basic_map/vector_data.rst]:
```suggestion
It is usually used to store discrete features, like roads and
```

File: source/docs/training_manual/basic_map/vector_data.rst [hunk 9]
```diff
@@ -8,9 +8,10 @@
...
+city blocks, Every object in a vector dataset is called a **feature**,
```
Comment (havatv) [inline source/docs/training_manual/basic_map/vector_data.rst]:
```suggestion
city blocks. The objects in a vector dataset are called **features**,
```

File: source/docs/training_manual/answers/answers.rst (hunk 4 above)
Comment (DelazJ) [reply source/docs/training_manual/answers/answers.rst]: Missing word in your suggestion @havatv ? 
Your sentence would be  ```.. tip:: A quicker approach could be to double-click the :guilabel:`rivers` layer, open the :menuselection:`Layer properties --> Source Fields` tab and count of fields.```
The idea I had is to simply take note of the max of rows as they are numbered in the table.

File: source/docs/training_manual/answers/answers.rst (hunk 6 above)
Comment (DelazJ) [reply source/docs/training_manual/answers/answers.rst]: This is just to skip me commenting each field (eg, IDs are xxx and zzz... with no great interest) of the table for those two features. But given that the instruction is "Tell us about towns" (with no limit), they are giving the opportunity to add any information they could take on those two rows.
Any suggestion of rewording?

File: source/docs/training_manual/basic_map/vector_data.rst [hunk 10]
```diff
@@ -8,9 +8,10 @@
...
+and is associated with data that describes that feature.
```
Comment (DelazJ) [inline source/docs/training_manual/basic_map/vector_data.rst]: I guess this end of sentence might also need update following @havatv reformulation

File: source/docs/training_manual/answers/answers.rst (hunk 4 above)
Comment (havatv) [reply source/docs/training_manual/answers/answers.rst]: If we are talking about rows, we can't use "fields".

File: source/docs/training_manual/answers/answers.rst (hunk 6 above)
Comment (havatv) [inline source/docs/training_manual/answers/answers.rst]:
```suggestion
  You can use other fields from these two records, if you like.
```

File: source/docs/training_manual/basic_map/vector_data.rst (hunk 10 above)
Comment (havatv) [inline source/docs/training_manual/basic_map/vector_data.rst]:
```suggestion
and are associated with data that describe their location and properties.
```

**Comment by havatv** [comment]
I have tried to address your comments.

File: source/docs/training_manual/answers/answers.rst (hunk 4 above)
Comment (tomchadwin) [reply source/docs/training_manual/answers/answers.rst]: But "rows" suggests records/features. How about "where you wil find a numbered list of [the table's] fields"?

File: source/docs/training_manual/answers/answers.rst [hunk 11]
```diff
@@ -58,6 +58,29 @@ Refer back to the :ref:`browser_panel_tm` and to :ref:`load_geopackage` sections
...
+  #. Select the layer in the :guilabel:`Layers` panel.
+  #. Right-click and choose :guilabel:`Attribute table`, or press the |openTable|
+     button on the :guilabel:`Attributes Toolbar`.
+  #. Count the number of columns.
+
+  .. tip:: A quicker approach could also consist in double-clicking the
+   :guilabel:`rivers` layer, open :menuselection:`Layer properties --> Source
+   Fields` tab and pick there the count of fields.
+
+* Information about towns are available in the :guilabel:`places` layer. Open its
```
Comment (tomchadwin) [inline source/docs/training_manual/answers/answers.rst]:
```suggestion
* Information about towns is available in the :guilabel:`places` layer. Open its
```

File: source/docs/training_manual/answers/answers.rst (hunk 5 above)
Comment (tomchadwin) [reply source/docs/training_manual/answers/answers.rst]:
```suggestion
  attribute table as you did with the :guilabel:`rivers` layer: there are two features whose :guilabel:`place`
```

File: source/docs/training_manual/basic_map/vector_data.rst (hunk 7 above)
Comment (tomchadwin) [reply source/docs/training_manual/basic_map/vector_data.rst]:
```suggestion
use of GIS. Vector data defines geographic features as points, lines and polygons on a
```

File: source/docs/training_manual/basic_map/vector_data.rst (hunk 10 above)
Comment (tomchadwin) [reply source/docs/training_manual/basic_map/vector_data.rst]:
```suggestion
and contain data that describe their location and properties.
```

File: source/docs/training_manual/basic_map/vector_data.rst [hunk 12]
```diff
@@ -22,24 +23,36 @@ It's important to know that the data you will be working with does not only
 represent **where** objects are in space, but also tells you **what** those
 objects are.
 
-From the previous exercise, you should have the :guilabel:`roads` layer
-loaded in your map. What you can see right now is merely the position of the
-roads.
+From the previous exercise, you should have the :guilabel:`rivers` layer
+loaded in your map. The lines that you can see right now is merely the
```
Comment (tomchadwin) [inline source/docs/training_manual/basic_map/vector_data.rst]:
```suggestion
loaded in your map. The lines that you can see right now are merely the
```

File: source/docs/training_manual/basic_map/vector_data.rst [hunk 13]
```diff
@@ -22,24 +23,36 @@ It's important to know that the data you will be working with does not only
...
+position of the rivers; this is the *spatial data*.
```
Comment (tomchadwin) [inline source/docs/training_manual/basic_map/vector_data.rst]:
```suggestion
position of the rivers: this is the *spatial data*.
```

File: source/docs/training_manual/basic_map/vector_data.rst [hunk 14]
```diff
@@ -22,24 +23,36 @@ It's important to know that the data you will be working with does not only
...
-To see all the data available to you, with the :guilabel:`roads` layer
+To see all the data available to you, with the :guilabel:`rivers` layer
 selected in the Layers panel, click on this button: |openTable|
 
-It will show you a table with more data about the :guilabel:`roads` layer.
-This extra data is called *attribute data*. The lines that you can see on your
-map represent where the roads go; this is the *spatial data*.
+It will show you a table with more data about the :guilabel:`rivers` layer.
+This extra dialog is the :guilabel:`Attribute table`. Each row, called
+*record*, represents a river *feature* and each column, also called *field*,
```
[the remark this answers is not in the document]
Comment (tomchadwin) [reply source/docs/training_manual/basic_map/vector_data.rst]: I like @havatv's suggestions, but would revert to "each" from the original.

File: source/docs/training_manual/basic_map/vector_data.rst [hunk 15]
```diff
@@ -22,24 +23,36 @@ It's important to know that the data you will be working with does not only
...
+represents a property of the river. Cells show *attributes*.
```
Comment (havatv) [inline source/docs/training_manual/basic_map/vector_data.rst]:
```suggestion
and represents a property of the river. Cells show *attributes*.
```

File: source/docs/training_manual/basic_map/vector_data.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/training_manual/basic_map/vector_data.rst]: For consistency I have updated the next line instead.

File: source/docs/training_manual/basic_map/vector_data.rst (hunk 10 above)
Comment (havatv) [reply source/docs/training_manual/basic_map/vector_data.rst]: Fine.

File: source/docs/training_manual/basic_map/vector_data.rst (hunk 7 above)
Comment (havatv) [reply source/docs/training_manual/basic_map/vector_data.rst]: @tomchadwin, I am not sure why this is an improvement. Could you explain your reason for suggesting this change?

File: source/docs/training_manual/answers/answers.rst (hunk 5 above)
Comment (havatv) [reply source/docs/training_manual/answers/answers.rst]: Fine.

File: source/docs/training_manual/answers/answers.rst (hunk 4 above)
Comment (havatv) [reply source/docs/training_manual/answers/answers.rst]: My mistake - I did not check the context. I have updated.

File: source/docs/training_manual/basic_map/vector_data.rst (hunk 7 above)
Comment (havatv) [reply source/docs/training_manual/basic_map/vector_data.rst]: What about "The 2D vector model" and "location and shape".

Comment (tomchadwin) [reply source/docs/training_manual/basic_map/vector_data.rst]: The reasons I suggested removing both of those are:

- I think the use of 2D could be confusing, given how many geometry types now supported in QGIS have a Z attribute
- I think the word "model" is unnecessary, and could potentially be confused with other meanings (such as "models" in 3D software)
- vector datasets don't only define location and shape, but also attributes

Comment (tomchadwin) [reply source/docs/training_manual/basic_map/vector_data.rst]: > points, lines and polygons (plus attributes) are very seldom exact representation of geographical objects. They are more or less accurate abstractions

A good point. How about in my suggestion changing "define" to "represent"? That allows for any levels of precision/abstraction, but doesn't limit it to geometry (as opposed to attribute data).

Comment (havatv) [reply source/docs/training_manual/basic_map/vector_data.rst]: Thank for the very valuable input! Let us see what @DelazJ thinks about this...

**Pull request merged**